enhancements/sig-policy/74-order-policy-execution/README.md

```diff
@@ -0,0 +1,219 @@
+policy level as well as an `extraDependencies` field on the template level to
+allow users to specify conditions for policy processing on both the policy and
+template level. When these fields are set and their conditions are not met, 
+the policy templates will not be applied on the managed cluster and the affected
+policy will enter a "pending" state. If this happens, any policy sets that contain
+the pending policy will also be marked as pending. When the conditions are met or
+the fields are not set, the policy templates will be processed normally.
+
+### User Stories
+
+1. As an author of a ConfigurationPolicy, I don’t want resources to be wasted when
+I know one object depends on another. If an object cannot be cleanly applied before
+another is created, I want to specify this in my Policy, and not waste
+resources (CPU, k8s API requests) in the policy controllers.
+2. As an author of a ConfigurationPolicy, I want to specify the order resources are
+applied in a policy when the API is not declarative. For some APIs (eg OLM), if
+resources are applied in the wrong order they can end up in an unhealthy state
+that I want to avoid.
+3. As a user of PolicySets, I want policies inside a set to be applied in a
+certain order.
+4. As an author of many ConfigurationPolicies, I have several policies that
+depend on another policy. I want to be able to reference completely separate
+policies (outside of the Policy “wrapper”) as required before a configuration
+is applied. (Maybe allow a policy to reference the status of a policy that runs
+before it (success or failure) before it tries to process it. Allowing you to chain
+them together)
+5. As a cluster administrator, I want to do a declarative action on a managed
+cluster when a policy becomes non-compliant. So I want to “activate” a
+ConfigurationPolicy based on the state of another policy in the cluster.
+6. As a user of the policy generator, I want to be able to generate policies that
+will have their templates be processed in a certain order.
+
+### Implementation Details/Notes/Constraints 
+
+We will leverage the new dynamic watcher library (enhancement 71) to watch all
+policy template dependencies. When a policy is created with a dependency, we
+can start to watch the policy object that it is dependent on. If the compliance
+of the dependency matches up with what the policy wants, we will add process it
+normally in the template sync. If not, we will assign a `Pending` state to the
+policy compliance, and the watcher library will notify us when the dependency
+changes, so we can re-evaluate the decision. This logic will be added into the
+new combined governance-policy-framework-addon repository.
+
+These dependencies will be able to accept any object that has a
+`status.complianceState` field - policies, policy sets, all current policy
+types, and anything that might use the policy nucleus in the future. If an
+object in `dependencies` or `extraDependencies` does not contain that status
```

Comment:
I'm not sure I like this because the Policy API has `omitempty` set on the `status.complianceState` field. This could lead to the weird situation where it goes from non-compliant to pending.

I would prefer to just keep it in pending and have a different status message based on if the field is set.